autoPyTorch/pipeline/components/setup/network/backbone/image.py

```diff
@@ -0,0 +1,229 @@
+from collections import OrderedDict
+from typing import Any, Dict, Optional, Tuple, List
+
+import ConfigSpace as CS
+import torch
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.hyperparameters import (
+    CategoricalHyperparameter,
+    UniformFloatHyperparameter,
+    UniformIntegerHyperparameter
+)
+from torch import nn
+from torch.nn import functional as F
+
+from autoPyTorch.pipeline.components.setup.network.backbone.base_backbone import BaseBackbone
+
+
+class ConvNetBackbone(BaseBackbone):
+    supported_tasks = {"image_classification", "image_regression"}
+
+    def _get_layer_size(self, w: int, h: int) -> Tuple[int, int]:
+        cw = ((w - self.config["conv_kernel_size"] + 2 * self.config["conv_kernel_padding"])
+              // self.config["conv_kernel_stride"]) + 1
+        ch = ((h - self.config["conv_kernel_size"] + 2 * self.config["conv_kernel_padding"])
+              // self.config["conv_kernel_stride"]) + 1
+        cw, ch = cw // self.config["pool_size"], ch // self.config["pool_size"]
+        return cw, ch
+
+    def _add_layer(self, layers: List[nn.Module], in_filters: int, out_filters: int) -> None:
+        layers.append(nn.Conv2d(in_filters, out_filters,
+                                kernel_size=self.config["conv_kernel_size"],
+                                stride=self.config["conv_kernel_stride"],
+                                padding=self.config["conv_kernel_padding"]))
+        layers.append(nn.BatchNorm2d(out_filters))
```

Comment:
Could we add an attribute to the networks `self.bn_args={"eps"=1e-05, "momentum"=0.1}` and have these be added when initializing bn layers `nn.BatchNorm2d(out_filters, **self.bn_args)`?

Reply:
Do you mean that they should be in the search space or just as a fixed attribute?

Reply:
I would set it as a fixed attribute for now. Perhaps in the future we want to study that influence or allow the user to change it manually

Reply:
added bn_args to the image networks